lxc/config/remote.go

```diff
@@ -52,20 +51,95 @@ func (c *Config) ParseRemote(raw string) (remoteName string, resourceName string
 
 // GetInstanceServer returns a lxd.InstanceServer for the remote with the given name.
 func (c *Config) GetInstanceServer(name string) (lxd.InstanceServer, error) {
+	return c.GetInstanceServerWithAdditionalConnectionArgs(name, nil)
+}
+
+// GetInstanceServerWithAdditionalConnectionArgs returns a lxd.InstanceServer for the remote with the given name. Any
+// populated fields of the given connection arguments override the default connection arguments for the remote.
+func (c *Config) GetInstanceServerWithAdditionalConnectionArgs(name string, inArgs *lxd.ConnectionArgs) (lxd.InstanceServer, error) {
```

Comment:
This function name seems excessively long, almost Java-esque.

How about just `GetInstanceServerWithConnectionArgs`, the "additional" is clear because the caller is passing them in.